src/database/DataTypes/RequisitionItem.js

```diff
@@ -148,6 +148,7 @@ export class RequisitionItem extends Realm.Object {
 
     // Return the maximum price of all MasterListItems.
     return UIDatabase.objects('MasterListItem')
+      .filtered('item.id == $0', this.item.id)
```

Comment:
This works, but I think there is a simpler solution. 

here: https://github.com/openmsupply/mobile/blob/de7f6813beb9e2b7874501bc2e28469102c74324/src/database/DataTypes/RequisitionItem.js#L145-L147 

The query already has `Item` - but we're adding it again here?

I *think* that the query is not correct. If you log the query I think it will look something like:

`masterList.id = X1 OR masterList.id = X2 OR, ... OR masterList.id = Xn AND item = Y1`

The evaluation of the conditions is incorrect - it would need to be one of any of the master lists or the last master list AND the item. We would want it to be anyone of the master lists AND the item, so just adding some parenthesis would do the trick:

`(masterList.id = X1 OR masterList.id = X2 OR, ... OR masterList.id = Xn) AND item = Y1`


So we can either add the parenthesis OR do what you have done but remove the `item` check in the above condition - up to you what you think is best, but we don't need it in both places

Reply:
You are right. It was working not applying a simpler solution as you mentioned (adding parenthesis).
Thanks for your feedback @joshxg!